tools/backend/new-lsq-generator/lsq-generator.py

```diff
@@ -0,0 +1,663 @@
+# This file generate the entire design of the new lsq, including:
+#   - Core LSQ design
+#   - Wrapper with extra peripheral logic for connecting the lsq
+import math
+import argparse
+import os
+import sys
+
+from configs import *
+from lsq_core import *
+
+#===----------------------------------------------------------------------===#
+# Parser Definition
+#===----------------------------------------------------------------------===#
+parser = argparse.ArgumentParser(description='Please specify the output path and lsq config file')
+parser.add_argument('--output-dir', '-o', dest='output_path', default='.', type=str)
+parser.add_argument('--config-file', '-c', required=True, dest='config_files', default='', type=str)
+
+# Build the target
+args = parser.parse_args()
+
+#===----------------------------------------------------------------------===#
+# Wrapper Generation
+#===----------------------------------------------------------------------===#
+
+class LSQWrapper:
+    """Class used to generate the top level wrapper for the LSQ
+    """
+    
+    def __init__(self, path_rtl: str, suffix: str, configs: Configs):
+        # Store the global information
+        self.output_folder = path_rtl
+        self.lsq_name = configs.name
+        self.module_suffix = suffix
+        self.lsq_config = configs
+        
+        # Define information needed for VHDL file generation
+        # This part is inherited from the design of the original lsq_generator
+        self.library_header = 'library IEEE;\nuse IEEE.std_logic_1164.all;\nuse IEEE.numeric_std.all;\n\n'
+        self.tab_level = 1
+        self.temp_count = 0
+        self.signal_init_str = ''
+        self.port_init_str = '\tport(\n\t\treset : in std_logic;\n\t\tclock : in std_logic'
+        self.reg_init_str = '\tprocess (clock, reset) is\n' + '\tbegin\n'
+        
+        # Define the final output string
+        self.lsq_wrapper_str = "\n\n"
+        
+    def genWrapper(self):
+        """This function generates the desired wrapper for the LSQ
+        """
+        
+        # PART 1: Add library information to the VHDL module
+        self.lsq_wrapper_str += self.library_header
+        
+        # PART 2: Define the entity
+        self.lsq_wrapper_str += f'entity {self.lsq_name} is\n'
+        
+        # PART 3: Add the module port definition
+        self.lsq_wrapper_str += self.port_init_str
+        
+        ##
+        ## Define all the IOs
```

Comment:
Here you can refer to the table that we plan to add at the beginning of the file for the documentation of the IO signals